src/routes/Batch.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState, FormEvent } from "react";
+import { useSelector } from "react-redux";
+import { getBatchById } from "../store/batches/batches.selectors";
+import { useParams } from "react-router-dom";
+import { getProductById } from "../store/products/products.selectors";
+import {
+  useFetchBatchIngredients,
+  useAddBatchIngredient
+} from "../store/batch-ingredients/batch-ingredients.hooks";
+import { useFetchBatches } from "../store/batches/batches.hooks";
+import { useFetchProducts } from "../store/products/products.hooks";
+import { BatchIngredient } from "../store/batch-ingredients/batch-ingredients.types";
+import { getIngredientById } from "../store/ingredients/ingredients.selectors";
+import format from "date-fns/format";
+import { useFetchIngredients } from "../store/ingredients/ingredients.hooks";
+import IngredientSelect from "../components/IngredientSelect";
+import { IngredientId } from "../store/ingredients/ingredients.types";
+import { BatchId } from "../store/batches/batches.types";
+import ingredients from "../store/ingredients/ingredients.reducer";
+
+interface BatchIngredientListItemProps {
+  batchIngredient: BatchIngredient;
+}
+const BatchIngredientListItem: React.FC<BatchIngredientListItemProps> = props => {
+  const { batchIngredient } = props;
+  const ingredient = useSelector(
+    getIngredientById(batchIngredient.ingredient_id)
+  );
+  if (!ingredient) {
+    return null;
+  }
+
+  return (
+    <li>
+      {ingredient.name} ({ingredient.type}), {batchIngredient.quantity_kg}
+      {ingredient.unit}
+    </li>
+  );
+};
+
+interface AddBatchIngredientProps {
+  batch_id: BatchId;
+}
+const AddBatchIngredient: React.FC<AddBatchIngredientProps> = props => {
+  const { batch_id } = props;
+  const [ingredientId, setIngredientId] = useState<IngredientId | undefined>(undefined);
+  const [quantity, setQuantity] = useState<number | undefined>(undefined);
+
+  const { submit: fetchBatchIngredients } = useFetchBatchIngredients(batch_id);
+  const { submit: addBatchIngredient, status } = useAddBatchIngredient();
+
+  const ingredient = useSelector(getIngredientById(ingredientId || -1));
+
+  const onSubmit = (e: FormEvent) => {
+    e.preventDefault();
+
+    if (ingredientId && quantity) {
+      addBatchIngredient({
+        batch_id,
+        ingredient_id: ingredientId,
+        quantity_kg: quantity
+      });
+    }
+  };
+
+  useEffect(() => {
+    if (status === "SUCCESS") {
+      setIngredientId(undefined);
+      setQuantity(undefined);
+      fetchBatchIngredients();
+    }
+  }, [status]);
+  return (
+    <form onSubmit={onSubmit}>
+      <IngredientSelect selected={ingredientId} onSelect={setIngredientId} />
+      <input
+        placeholder={`Quantity ${ingredient ? `(${ingredient.unit})` : ""}`}
```

Comment:
where does the text 'Select an ingredient' that appears on the dropdown for adding a batch come in? I thought it might be here... Am I correct that it is this file that creates the functionality to add the ingredients to a batch?

Reply:
> Am I correct that it is this file that creates the functionality to add the ingredients to a batch?
Exactly
> where does the text 'Select an ingredient' ...
i believe it's in `IngredientSelect`